### PR TITLE
INFRA-164: GHA runner OS version pinning

### PR DESCRIPTION
**Pull request by BenedekFarkas**
[INFRA-164](https://lombiq.atlassian.net/browse/INFRA-164)

### CODE REVIEW

File: .readthedocs.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@
 version: 2
 
 build:
-  os: ubuntu-22.04
+  os: ubuntu-24.04
```
Comment (Piedone) [inline .readthedocs.yml:8]: How do you test this?

**Comment by Piedone** [comment]
If you can somehow run the ReadTheDocs build and it passes, then feel free to merge.

File: .readthedocs.yml (hunk 1 above)
Comment (BenedekFarkas) [reply .readthedocs.yml:8]: I don't have access to it, @sarahelsaig can you please check? If it's a hassle, we can just omit this change, it's not that important for the docs.

Comment (sarahelsaig) [reply .readthedocs.yml:8]: I don't know. @agriffard ?

Comment (Piedone) [reply .readthedocs.yml:8]: I just remembered, and yes, we need Antoine's help here. @BenedekFarkas @sarahelsaig please sign up with your Lombiq addresses here: https://readthedocs.org/dashboard/

Then, Antoine, please add benedek.farkas@lombiq.com, sara.el-saig@lombiq.com, and zoltan.lehoczky@lombiq.com as maintainers.

FYI for OC, there's an OC project there, the admin of which you can access under https://readthedocs.org/projects/orchardcore/. Bence, once you sign up with, I can add you there too.

Once done, I _think_ you can add a new version under https://readthedocs.org/projects/orchardcore/versions/ for the branch and build the docs only for that.

The whole thing should be analogous for OCC too.

Comment (agriffard) [reply .readthedocs.yml:8]: zoltan.lehoczky@lombiq.com added as maintainer.
benedek.farkas@lombiq.com and sara.el-saig@lombiq.com don't currently exist as user on RTD, tell me when I can add them.

Comment (Piedone) [reply .readthedocs.yml:8]: Ah, there's an invitation e-mail. Accepted it, now it works.

I activated the version and a build is now running.

Comment (Piedone) [reply .readthedocs.yml:8]: Works: https://commerce.orchardcore.net/en/issue-infra-164/ I now deactivated it.

Nevertheless, Bence, Sári, please still sign up to RTD.

**Review by Piedone — APPROVED** [review]

File: .readthedocs.yml (hunk 1 above)
Comment (sarahelsaig) [reply .readthedocs.yml:8]: I've signed up now.

Comment (Piedone) [reply .readthedocs.yml:8]: Added you.

Comment (BenedekFarkas) [reply .readthedocs.yml:8]: I've just signed up as well.

Comment (Piedone) [reply .readthedocs.yml:8]: Added you too.

Comment (BenedekFarkas) [reply .readthedocs.yml:8]: I'm in, thanks!

**Pull request merged**